substratevm/src/com.oracle.svm.hosted/src/com/oracle/svm/hosted/InstrumentFeature.java

```diff
@@ -94,12 +97,32 @@ public void afterRegistration(AfterRegistrationAccess access) {
      * is absent. <br>
      * So this method looks for them in the same order.
      */
-    private void addPremainClass(String premainClass) {
+    private void addPremainClass(String javaagentOption) {
+        int separatorIndex = javaagentOption.indexOf("=");
+        String agent;
+        String premainClass = null;
+        String options = "";
+        // Get the agent file
+        if (separatorIndex == -1) {
+            agent = javaagentOption;
+        } else {
+            agent = javaagentOption.substring(0, separatorIndex);
+            options = javaagentOption.substring(separatorIndex + 1);
+        }
+        // Read MANIFEST in agent jar
+        try {
+            JarFile agentJarFile = new JarFile(agent);
+            premainClass = agentJarFile.getManifest().getMainAttributes().getValue("Premain-Class");
+        } catch (IOException e) {
+            UserError.abort(e, "Can't read the agent jar %s. Please check option %s", agent,
+                            SubstrateOptionsParser.commandArgument(SubstrateOptions.JavaAgent, ""));
+        }
+
         try {
             Class<?> clazz = Class.forName(premainClass, false, cl);
             Method premain = null;
             List<Object> args = new ArrayList<>();
-            args.add(""); // First argument is options which will be set at runtime
+            args.add(options); // First argument is options which will be set at runtime
```

Comment:
Can the options be overridden at run time?

Reply:
Yes. Options set at runtime can override the build time options.